packages/p2p/src/stores/general-store.js

```diff
@@ -220,10 +222,18 @@ export default class GeneralStore extends BaseStore {
                 } else if (is_fully_authed_and_does_not_need_fa) {
                     this.setIsLoading(false);
                 }
+            } else if (!error && get_account_status.risk_classification === 'low') {
+                const is_blocked_for_pa = hasStatuses(['p2p_blocked_for_pa']);
+
+                if (is_blocked_for_pa) {
+                    this.setIsPaymentAgentBlocked(true);
+                    this.setIsLoading(false);
```

Comment:
There seems to be else blocks where `this.setIsLoading(false);` is repeated. Please check if they can be refactored to remove the function call in each block